msg/templates/uorb_microcdr/microRTPS_client.cpp.em

```diff
@@ -166,6 +196,49 @@ void *send(void *args)
 		}
 @[    end for]@
 
+@[    if send_topics_poll]@
+		{
+			px4_pollfd_struct_t fds[@(len(send_topics_poll))];
+@[        for idx, topic in enumerate(send_topics_poll)]@
+			fds[@(idx)].fd = subs_poll->@(topic)_sub;
+			fds[@(idx)].events = POLLIN;
+@[        end for]@
+
+			int pret = px4_poll(fds, @(len(send_topics_poll)), 20);
+
+			if (pret < 0) {
+				PX4_ERR("poll failed (%i)", pret);
+			} else if (pret != 0) {
+@[        for idx, topic in enumerate(send_topics_poll)]@
+				if (fds[@(idx)].revents & POLLIN) {
+					@(send_base_types_poll[idx])_s @(topic)_data;
+					orb_copy(ORB_ID(@(topic)), subs_poll->@(topic)_sub, &@(topic)_data);
+					serialize_@(send_base_types_poll[idx])(&writer, &@(topic)_data, &data_buffer[header_length], &length);
+
+					if (0 < (read = transport_node->write(static_cast<char>(@(msgs[0].index(topic) + 1)), data_buffer, length))) {
+						data->total_sent += read;
+						tx_last_sec_read += read;
+						++data->sent;
+					}
+				}
+@[        end for]@
+			} else {
+				if (hrt_absolute_time() - last_stats_update >= 1_s) {
```

Comment:
This should be checked in any case, not only on poll timeout.
Also can you avoid the code duplication and move this to a separate function (or move it outside of `@[ if send_topics_poll]@`)?

Reply:
Done